File: Blog/Cloud/AWS/src/packaging.02/router.js
'use strict';

const _ = require('lodash');

const routeMap = {
  '/test': {
    'GET': (event, context) => {
      const userId = _.get(event,'queryStringParameters.id');
      return require('./controllers/test/get')(userId);
    },
    'POST': (event, context) => {
      const userId = _.get(event,'queryStringParameters.id');
      const body = JSON.stringify(_.get(event,'body'));
      const header =  event.headers;
      return require('./controllers/test/post')(userId, header, body);
    }
  }
};

module.exports = (event, context) => {
  const controller = routeMap[event.path][event.httpMethod];

  if(!controller) {
    return {
      body: { Error: "Invalid Path" }
    };
  }

  return controller(event, context);
};

File: Blog/Cloud/AWS/src/Lambda.Packaging.Python/router.py
import controllers.test.get
import controllers.test.post

route_map = {
    '/test': {
        'GET': controllers.test.get.handler,
        'POST': controllers.test.post.handler
    }
};

def router(event):
    controller = route_map[event['path']][event['httpMethod']];
    
    if not controller:
        return { 'body': { 'Error': "Invalid Path" } }
    
    return controller(event);